pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -624,11 +624,11 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 }
 
 func (r *ReconcileClusterDeployment) reconcileInstallingClusterProvision(cd *hivev1.ClusterDeployment, releaseImage string, logger log.FieldLogger) (reconcile.Result, error) {
-	// Return early and stop processing if Agent install strategy is in play. The controllers that
+	// Return early and stop processing if ClusterInstallRef is in play. The controllers that
 	// handle this portion of the API currently live in the assisted service repo, rather than hive.
 	// This will hopefully change in the future.
-	if cd.Spec.Provisioning != nil && cd.Spec.Provisioning.InstallStrategy != nil && cd.Spec.Provisioning.InstallStrategy.Agent != nil {
-		logger.Info("skipping processing of agent install strategy cluster")
+	if cd.Spec.ClusterInstallRef != nil {
+		logger.Info("ClusterInstallRef specified, skipping creation of ClusterProvision")
 		return reconcile.Result{}, nil
 	}
```

Comment:
i don't think we need to add duplicate check for clusterinstallref here, the function already assumes that it was invoked for clusterprovision